core/src/main/java/hudson/util/TextFile.java

```diff
@@ -81,34 +146,8 @@ public String read() throws IOException {
     /**
      * Parse text file line by line.
      */
-    public Iterable<String> lines() {
-        return new Iterable<String>() {
-            @Override
-            public Iterator<String> iterator() {
-                try {
-                    final BufferedReader in = new BufferedReader(new InputStreamReader(
-                            Files.newInputStream(file.toPath()),"UTF-8"));
-
-                    return new AbstractIterator<String>() {
-                        @Override
-                        protected String computeNext() {
-                            try {
-                                String r = in.readLine();
-                                if (r==null) {
-                                    in.close();
-                                    return endOfData();
-                                }
-                                return r;
-                            } catch (IOException e) {
-                                throw new RuntimeException(e);
-                            }
-                        }
-                    };
-                } catch (IOException | InvalidPathException e) {
-                    throw new RuntimeException(e);
-                }
-            }
-        };
+    public @Nonnull LinesStream lines() throws IOException {
```

Comment:
The exception needs to be documented at least

Reply:
Done